File: NIM.py
def prikazi(gomile):
    print("\nGomile:")
    for i, x in enumerate(gomile):
        print(f"{i+1}: {'|' * x} ({x})")


def nim_sum(gomile):
    s = 0
    for x in gomile:
        s ^= x
    return s


while True:

    n = int(input("Broj gomila: "))
    gomile = []

    for i in range(n):
        gomile.append(
            int(input(f"Gomila {i+1}: "))
        )

    while sum(gomile) > 0:

        prikazi(gomile)

        g = int(input("\nIzaberi gomilu: ")) - 1
        k = int(input("Koliko uzimas: "))

        gomile[g] -= k

        if sum(gomile) == 0:
            print("\nPobedio si!")
            break

        x = nim_sum(gomile)

        if x == 0:
            for i in range(n):
                if gomile[i] > 0:
                    gomile[i] -= 1
                    print(
                        f"\nRacunar uzima 1 iz gomile {i+1}"
                    )
                    break
        else:
            for i in range(n):
                target = gomile[i] ^ x

                if target < gomile[i]:
                    take = gomile[i] - target
                    gomile[i] = target

                    print(
                        f"\nRacunar uzima {take} iz gomile {i+1}"
                    )
                    break

        if sum(gomile) == 0:
            print("\nRacunar je pobedio!")
            break

    break

File: NIM.cpp
#include <bits/stdc++.h>
using namespace std;

int nimSum(vector<int>& piles) {
    int s = 0;
    for (int x : piles)
        s ^= x;
    return s;
}

void printPiles(vector<int>& piles) {
    cout << "\nGomile:\n";
    for (int i = 0; i < piles.size(); i++)
        cout << i + 1 << ": " << piles[i] << endl;
}

bool gameOver(vector<int>& piles) {
    for (int x : piles)
        if (x > 0) return false;
    return true;
}

int main() {
    int n;
    cout << "Broj gomila: ";
    cin >> n;

    vector<int> piles(n);

    for (int i = 0; i < n; i++) {
        cout << "Elementi u gomili " << i + 1 << ": ";
        cin >> piles[i];
    }

    while (true) {

        printPiles(piles);

        int pile, take;
        cout << "\nTvoj potez\n";
        cout << "Gomila: ";
        cin >> pile;

        cout << "Koliko uzimas: ";
        cin >> take;

        piles[pile - 1] -= take;

        if (gameOver(piles)) {
            cout << "\nPobedio si!\n";
            break;
        }

        int x = nimSum(piles);

        if (x == 0) {
            for (int i = 0; i < n; i++) {
                if (piles[i] > 0) {
                    piles[i]--;
                    cout << "\nRacunar uzima 1 iz gomile "
                         << i + 1 << endl;
                    break;
                }
            }
        }
        else {
            for (int i = 0; i < n; i++) {
                int target = piles[i] ^ x;

                if (target < piles[i]) {
                    int take = piles[i] - target;
                    piles[i] = target;

                    cout << "\nRacunar uzima "
                         << take
                         << " iz gomile "
                         << i + 1 << endl;
                    break;
                }
            }
        }

        if (gameOver(piles)) {
            cout << "\nRacunar je pobedio!\n";
            break;
        }
    }
}
